src/cp/cp_api.c

```diff
@@ -0,0 +1,368 @@
+{
+    WT_CONTROL_POINT *saved_data;
+
+    __wt_spin_lock(session, &cp_registry->lock);
+    saved_data = cp_registry->data;
+    if (saved_data != NULL)
+        __wt_atomic_add32(&saved_data->ref_count, 1);
+    if (!locked)
+        __wt_spin_unlock(session, &cp_registry->lock);
+    return (saved_data);
+}
+
+/*
+ * __wt_control_point_unlock --
+ *     Unlock after additional processing.
+ *
+ * This is called after finishing the additional processing started with
+ *     __wti_control_point_get_data() with locked=true.
+ *
+ * @param session The session. @param cp_registry The control point registry.
+ */
+void
+__wt_control_point_unlock(WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry)
+{
+    __wt_spin_unlock(session, &cp_registry->lock);
+}
+
+/*
+ * __wt_control_point_release_data --
+ *     Call when done using WT_CONTROL_POINT_REGISTRY->data that was returned by
+ *     __wti_control_point_get_data.
+ *
+ * @param session The session. @param cp_registry The control point registry. @param locked True if
+ *     the control point data is already locked.
+ */
+void
+__wt_control_point_release_data(WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry,
+  WT_CONTROL_POINT *data, bool locked)
+{
+    uint32_t new_ref;
+
+    if (WT_UNLIKELY(data == NULL))
+        return;
+    if (!locked)
+        __wt_spin_lock(session, &cp_registry->lock);
+    new_ref = __wt_atomic_sub32(&cp_registry->data->ref_count, 1);
+    if ((new_ref == 0) && (cp_registry->data != data))
+        __wt_free(session, data);
+    __wt_spin_unlock(session, &cp_registry->lock);
+}
+
+/*
+ * API: Disable a per connection control point.
+ */
+/*
+ * __conn_control_point_disable --
+ *     Disable a per connection control point given a WT_CONTROL_POINT_REGISTRY.
+ *
+ * @param session The session. @param cp_registry The WT_CONTROL_POINT_REGISTRY of the per
+ *     connection control point to disable.
+ */
+static int
+__conn_control_point_disable(WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry)
+{
+    WT_CONTROL_POINT *saved_data;
+    WT_DECL_RET;
+
+    __wt_spin_lock(session, &cp_registry->lock);
+    saved_data = cp_registry->data;
+    if (WT_UNLIKELY(saved_data == NULL))
+        /* Already disabled. */
+        WT_ERR(WT_NOTFOUND);
+    cp_registry->data = NULL;
+    if (__wt_atomic_loadv32(&saved_data->ref_count) == 0)
+        __wt_free(session, saved_data);
+#if 0 /* TODO. */
+    else
+        /* TODO: Add saved_data to a queue of control point data waiting to be freed. */;
+#endif
+err:
+    __wt_spin_unlock(session, &cp_registry->lock);
+    return (ret);
+}
+
+/*
+ * __wt_conn_control_point_disable --
+ *     Disable a per connection control point.
+ *
+ * @param wt_session The session. @param id The ID of the per connection control point to disable.
+ */
+int
+__wt_conn_control_point_disable(WT_SESSION *wt_session, wt_control_point_id_t id)
```

Comment:
This and the enable function will need to be connection level API's eventually. e.g. conn->control_point_disable. Did we do this way for the POC?

Reply:
I just forgot to do it. Jie pointed it out. I did it today.
Done